npm error ERESOLVE unable to resolve dependency tree when using npm package react@rc

```
    ~/repos/nextjs-03  on   main  npm i @floating-ui/react                                                                   ✔  at 19:52:19  
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: nextjs-03@0.1.0
npm error Found: react@19.0.0-rc-66855b96-20241106
npm error node_modules/react
npm error   react@"19.0.0-rc-66855b96-20241106" from the root project
npm error
npm error Could not resolve dependency:
npm error peer react@">=16.8.0" from @floating-ui/react@0.26.28
npm error node_modules/@floating-ui/react
npm error   @floating-ui/react@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /Users/xyz/.npm/_logs/2024-12-02T18_52_29_768Z-eresolve-report.txt
npm error A complete log of this run can be found in: /Users/xyz/.npm/_logs/2024-12-02T18_52_29_768Z-debug-0.log```

Getting the above error when trying to install floating-ui in my Next.js 15 app.

Repro:

1. npx create-next-app@latest
2. npm i @floating-ui/react

--- a/.changeset/plenty-pets-call.md
+++ b/.changeset/plenty-pets-call.md
@@ -1,0 +1,5 @@
+---
+'@floating-ui/react': minor
+---
+
+chore: drop React 16 support. 17 is the minimum supported version.

--- a/.changeset/plenty-pets-call.md
+++ b/.changeset/plenty-pets-call.md
@@ -2,4 +2,4 @@
 '@floating-ui/react': minor
 ---
 
-chore: drop React 16 support. 17 is the minimum supported version.
+breaking: drop React 16 support. 17 is the minimum supported version.

--- a/.changeset/pretty-pots-deny.md
+++ b/.changeset/pretty-pots-deny.md
@@ -1,0 +1,6 @@
+---
+"@floating-ui/react-dom": patch
+"@floating-ui/react": patch
+---
+
+chore: add React 19 RC support

--- a/.changeset/pretty-pots-deny.md
+++ b/.changeset/pretty-pots-deny.md
@@ -1,6 +1,7 @@
 ---
-"@floating-ui/react-dom": patch
-"@floating-ui/react": patch
+'@floating-ui/react-dom': patch
+'@floating-ui/react': patch
+'@floating-ui/react-native': patch
 ---
 
 chore: add React 19 RC support

--- a/packages/react-dom/package.json
+++ b/packages/react-dom/package.json
@@ -57,8 +57,8 @@
     "react-dom"
   ],
   "peerDependencies": {
-    "react": ">=16.8.0",
-    "react-dom": ">=16.8.0"
+    "react": ">=16.8.0 || ^19.0.0-rc",
+    "react-dom": ">=16.8.0 || ^19.0.0-rc"
   },
   "dependencies": {
     "@floating-ui/dom": "^1.0.0"

--- a/packages/react-native/package.json
+++ b/packages/react-native/package.json
@@ -42,7 +42,7 @@
     "react-native"
   ],
   "peerDependencies": {
-    "react": ">=16.8.0",
+    "react": ">=16.8.0 || ^19.0.0-rc",
     "react-native": ">=0.64.0"
   },
   "dependencies": {

--- a/packages/react/package.json
+++ b/packages/react/package.json
@@ -68,8 +68,8 @@
     "react-dom"
   ],
   "peerDependencies": {
-    "react": ">=16.8.0",
-    "react-dom": ">=16.8.0"
+    "react": ">=17.0.0",
+    "react-dom": ">=17.0.0"
   },
   "dependencies": {
     "@floating-ui/react-dom": "workspace:^",

--- a/packages/react/package.json
+++ b/packages/react/package.json
@@ -68,8 +68,8 @@
     "react-dom"
   ],
   "peerDependencies": {
-    "react": ">=17.0.0",
-    "react-dom": ">=17.0.0"
+    "react": ">=17.0.0 || ^19.0.0-rc",
+    "react-dom": ">=17.0.0 || ^19.0.0-rc"
   },
   "dependencies": {
     "@floating-ui/react-dom": "workspace:^",

